### PR TITLE
fix(dropdown): integrate component audit feedback

### LINKED ISSUE #1792: Component Design Audit v1.1 - Dropdown
References: #1749 

## Dropdown

- Update padding to 16px on left/right, vertically centered text

image

- Needs hover state

- Rulers need to not show on hover

image

- When open, label should change color to match spec

image

- focus outline should be against the sides of the menu

image

- remove bottom margin from label when helper text is present

image

- should return back to field-01 when not hovered

image

- should keep field-01 rule when open

image

- border bottom color should persist (gray-20, not gray-50) ui-03

image

- show selected state (selected-ui token)

image

- busted focus (react)

image

- change height to 320px(ish)

- look into shadow colors

### Inline

- No bar should be present

image

- should have hover state

- 15px on left-hand side

image

- height should be 32px

- text alignment between field and options

### PR DESCRIPTION
**Pull request by emyarod**
References #1749

Closes #1792

This PR aims to align the experimental dropdown component with the latest component audit feedback.

#### Changelog

**Changed**

- spacing and color based on component audit feedback
- inline dropdown is now actually inline
- decouple invalid icon styles from data attribute

### CODE REVIEW

File: src/components/dropdown/_dropdown.scss [hunk 1]
```diff
@@ -309,8 +321,8 @@
     height: rem(40px);
     padding-top: rem(12px);
     padding-bottom: rem(12px);
-    padding-left: rem(15px);
-    padding-right: $spacing-2xl;
+    padding-left: 1rem;
```
Comment (asudoh) [inline src/components/dropdown/_dropdown.scss]:
```suggestion
    padding-left: rem(16px);
```

File: src/components/dropdown/_dropdown.scss [hunk 2]
```diff
@@ -363,16 +375,16 @@
     font-weight: normal;
     line-height: rem(16px);
     padding: rem(11px) 0;
-    margin: 0 rem(16px);
+    margin: 0 1rem;
```
Comment (asudoh) [inline src/components/dropdown/_dropdown.scss]:
```suggestion
    margin: 0 rem(16px);
```

File: src/components/dropdown/_dropdown.scss [hunk 3]
```diff
@@ -363,16 +375,16 @@
...
     border-bottom: 1px solid $ui-03;
     text-overflow: ellipsis;
     overflow: hidden;
 
     &:focus {
       @include focus-outline('outline');
-      margin: 0 rem(2px);
-      padding: rem(11px) rem(14px);
+      margin: 0;
+      padding: rem(11px) 1rem;
```
Comment (asudoh) [inline src/components/dropdown/_dropdown.scss]:
```suggestion
      padding: rem(11px) rem(16px);
```

File: src/components/dropdown/_dropdown.scss [hunk 4]
```diff
@@ -419,20 +431,26 @@
...
     }
 
     &:hover {
-      background-color: transparent;
+      background-color: $hover-ui;
     }
   }
 
+  .#{$prefix}--dropdown--inline.#{$prefix}--dropdown--invalid .#{$prefix}--dropdown-text__inner {
+    margin-right: 1.25rem;
```
Comment (asudoh) [inline src/components/dropdown/_dropdown.scss]:
```suggestion
    margin-right: rem(20px);
```

**Review by asudoh — COMMENTED** [review]
LGTM basically, just one thing - I believe we are using `rem()` function with pixel values, though there are some exceptions. I see many of dropdown code is not in such exception. Thanks!

**Comment by tw15egan** [comment]
Not sure what's causing it, but I'm getting this error when checking out this PR and trying to switch to experimental styles:

screen shot 2019-02-14 at 11 52 16 am

**Comment by emyarod** [comment]
thanks @tw15egan should be fixed now

**Review by tw15egan — APPROVED** [review]
Looks good now!

**Review by jnm2377 — CHANGES_REQUESTED** [review]
I'm seeing this weird bug on hover. Both locally and in netlify. 
The lines disappear if you're hovering anywhere on the dropdown.

dropdown

**Comment by emyarod** [comment]
@jnm2377 that was my interpretation of the component audit feedback

> * Rulers need to not show on hover

cc @IBM/carbon-designers for clarification

**Comment by tw15egan** [comment]
I take it as the currently selected item loses the divider on hover, like how it currently is:

https://next.carbondesignsystem.com/components/dropdown/code

**Comment by emyarod** [comment]
isn't that due to the similarities between the border and hover colors? unless I am misunderstanding

**Comment by tw15egan** [comment]
I feel like hiding all of the dividers when hovering over the entire dropdown is a pretty jarring effect.

I think they are referring to the 1 px divider that is still being shown like here: 
screen shot 2019-02-14 at 4 15 14 pm

**Comment by emyarod** [comment]
changed it to hide only the hovered element's border

**Comment by tw15egan** [comment]
Looks great, thanks!

**Comment by asudoh** [comment]
@jnm2377 Does @emyarod's latest code look good to you...? Thanks!

**Review by jnm2377 — APPROVED** [review]
Looks great! 🎉

**Pull request merged**